pkg/cmd/issue/edit/edit.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	shared "github.com/cli/cli/pkg/cmd/issue/shared"
+	prShared "github.com/cli/cli/pkg/cmd/pr/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+)
+
+type EditOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	OpenInBrowser      func(string) error
+	DetermineEditor    func() (string, error)
+	FieldsToEditSurvey func(*prShared.EditableOptions) error
+	EditableSurvey     func(string, *prShared.EditableOptions) error
+	FetchOptions       func(*api.Client, ghrepo.Interface, *prShared.EditableOptions) error
+
+	SelectorArg string
+	Interactive bool
+	WebMode     bool
+
+	prShared.EditableOptions
+}
+
+func NewCmdEdit(f *cmdutil.Factory, runF func(*EditOptions) error) *cobra.Command {
+	opts := &EditOptions{
+		IO:                 f.IOStreams,
+		HttpClient:         f.HttpClient,
+		OpenInBrowser:      utils.OpenInBrowser,
+		DetermineEditor:    func() (string, error) { return cmdutil.DetermineEditor(f.Config) },
+		FieldsToEditSurvey: prShared.FieldsToEditSurvey,
+		EditableSurvey:     prShared.EditableSurvey,
+		FetchOptions:       prShared.FetchOptions,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "edit {<number> | <url>}",
+		Short: "Edit an issue",
+		Example: heredoc.Doc(`
+			$ gh issue edit 23 --title "I found a bug" --body "Nothing works"
+			$ gh issue edit 23 --label "bug,help wanted"
+			$ gh issue edit 23 --label bug --label "help wanted"
+			$ gh issue edit 23 --assignee monalisa,hubot
+			$ gh issue edit 23 --assignee @me
+			$ gh issue edit 23 --project "Roadmap"
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.SelectorArg = args[0]
+
+			flags := cmd.Flags()
+			if flags.Changed("title") {
+				opts.EditableOptions.TitleEdited = true
+			}
+			if flags.Changed("body") {
+				opts.EditableOptions.BodyEdited = true
+			}
+			if flags.Changed("assignee") {
+				opts.EditableOptions.AssigneesEdited = true
+			}
+			if flags.Changed("label") {
+				opts.EditableOptions.LabelsEdited = true
+			}
+			if flags.Changed("project") {
+				opts.EditableOptions.ProjectsEdited = true
+			}
+			if flags.Changed("milestone") {
+				opts.EditableOptions.MilestoneEdited = true
+			}
+
+			if !opts.EditableOptions.Dirty() {
+				opts.Interactive = true
+			}
+
+			if opts.Interactive && !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("--tile, --body, --assignee, --label, --project, --milestone, or --web required when not running interactively")}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return editRun(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open the browser to create an issue")
```

Comment:
I propose that instead of providing the `--web` option here we defer to people using `gh issue view --web` as it was possible before. I do not see any value that `gh issue edit --web` adds over the previous command, only potential confusion.